### PR TITLE
Update to eslint

### PR DESCRIPTION
**Pull request by mixonic**
I'm taking a few iterative steps at upgrading the build here, hopefully I can smooth the path for 3.0 support and general contribution.

This PR:

* Pin a bower dependency that does not have the correct assets unless it is pinned
* Updates the project to use eslint
* Updates several files to match best practices
* Drops PhantomJS from testing
* Tweaks the `.travis.yml` to be closed to Ember CLI master
* Bump uglify, ember-load-initializers, loader.js

### CODE REVIEW

**Comment by Gaurav0** [comment]
Thanks for working on this!

**Comment by mixonic** [comment]
Finally. I could not figure out why the CI suite was crashing Chrome on Travis. It turns out the testem.js config I took from Ember CLI's latest blueprint is incompatible with the version of Testem currently locked as a dependency, and as a result Chrome was not running in a headless mode. I've written a slightly bespoke Testem config that is headless on Travis, which seems sufficient.

Onward!

File: package.json [hunk 1]
```diff
@@ -74,11 +72,15 @@
     "ember-route-action-helper": "^1.0.0",
     "ember-source": "~2.11.0",
     "ember-test-helpers": "~0.6.3",
+    "eslint-plugin-ember": "^5.0.0",
```
Comment (Gaurav0) [inline package.json]: Please use a ~ here. New minor versions of `eslint-plugin-ember` tend to introduce new rules which cause builds to fail.

**Comment by Gaurav0** [comment]
Canary is broken.

image

**Comment by Gaurav0** [comment]
The PR appears to work locally.

**Pull request merged**